cdap-ui/app/cdap/components/Replicator/Detail/index.tsx

```diff
@@ -0,0 +1,190 @@
+  const [replicator, setReplicator] = React.useState(Map());
+  const [error, setError] = React.useState();
+  const [status, setStatus] = React.useState();
+  const [redirect, setRedirect] = React.useState(false);
+
+  function getBaseParams() {
+    return {
+      namespace: getCurrentNamespace(),
+      appName: match.params.replicatorId,
+    };
+  }
+
+  React.useEffect(() => {
+    MyReplicatorApi.getReplicator(getBaseParams()).subscribe((app) => {
+      let config;
+
+      try {
+        config = JSON.parse(app.configuration);
+      } catch (e) {
+        setError(e);
+      }
+
+      setReplicator(fromJS(config));
+    });
+
+    MyReplicatorApi.pollStatus(getBaseParams()).subscribe((res) => {
+      setStatus(res.status);
+    });
+  }, []);
+
+  function start() {
+    const params = {
+      ...getBaseParams(),
+      action: 'start',
+    };
+
+    MyReplicatorApi.action(params).subscribe(fetchStatus, (err) => {
+      setError(err);
+    });
+  }
+
+  function stop() {
+    const params = {
+      ...getBaseParams(),
+      action: 'stop',
+    };
+
+    MyReplicatorApi.action(params).subscribe(fetchStatus, (err) => {
+      setError(err);
+    });
+  }
+
+  function deleteReplicator() {
+    MyReplicatorApi.delete(getBaseParams()).subscribe(
+      () => {
+        setRedirect(true);
+      },
+      (err) => {
+        setError(err);
+      }
+    );
+  }
+
+  function fetchStatus() {
+    MyReplicatorApi.getStatus(getBaseParams()).subscribe((res) => {
+      setStatus(res.status);
+    });
+  }
+
+  const listViewLink = `/ns/${getCurrentNamespace()}/replicator`;
+
+  if (redirect) {
+    return <Redirect to={listViewLink} />;
+  }
+
+  const startTime = moment()
+    .subtract(7, 'days')
+    .format('X');
+  let logUrl = `/v3/namespaces/${getCurrentNamespace()}/apps/${
+    match.params.replicatorId
+  }/workers/DeltaWorker/logs`;
+
+  logUrl = `${logUrl}?start=${startTime}`;
+  logUrl = `/downloadLogs?type=raw&backendPath=${encodeURIComponent(logUrl)}`;
+
+  return (
+    <div className={classes.root}>
+      <div>
+        <Link to={listViewLink}>Back to List View</Link>
+      </div>
+      <h2>{match.params.replicatorId}</h2>
+      <br />
+
+      <If condition={error}>
+        <div className="text-danger">
+          <strong>{JSON.stringify(error, null, 2)}</strong>
+          <br />
+        </div>
+      </If>
+      <div>
+        <Status status={status} />
+      </div>
+      <br />
+      <div className={classes.buttonContainer}>
+        <Button variant="contained" color="primary" onClick={start} disabled={status === 'RUNNING'}>
+          Start
+        </Button>
+
+        <Button variant="outlined" onClick={stop} disabled={status === 'STOPPED'}>
+          Stop
+        </Button>
+
+        <Button variant="outlined" color="primary" href={logUrl} target="_tab">
+          Logs
+        </Button>
+
+        <Button variant="contained" color="secondary" onClick={deleteReplicator}>
+          DELETE
```

Comment:
Why is DELETE in all caps?

Reply:
this detail view is a temporary placeholder